sso-kit-client/lit/src/SingleSignOnContext.ts

```diff
@@ -7,13 +7,31 @@
  * See <https://vaadin.com/commercial-license-and-service-terms> for the full
  * license.
  */
-import { logout } from "@hilla/frontend";
 import type { Subscription } from "@hilla/frontend";
+import { logout } from "@hilla/frontend";
+import type {
+  ActionFn,
+  ActionResult,
+  Commands,
+  Context,
+  Route,
+} from "@vaadin/router";
 import type { AccessProps } from "../../core/src/AccessProps.js";
 import type { SingleSignOnData } from "../../core/src/SingleSignOnData.js";
 import type { User } from "../../core/src/User.js";
 import { EndpointImportError } from "../../core/src/EndpointImportError.js";
 
+/**
+ * Definition of the route extended with {@link AccessProps} used to define a
+ * view route.
+ */
+type ViewRoute = Route &
+  AccessProps & {
+    title?: string;
+    icon?: string;
+    children?: ViewRoute[];
+  };
```

Comment:
I'd rename this to `ProtectedRoute`:
```suggestion
type ProtectedRoute = Route & AccessProps & {
  title?: string;
  icon?: string;
  children?: ProtectedRoute[];
};
```

Also, why we add `title`, `icon` and `children`? Doesn't `Route` already have those? If not, why should we add them?